### PR TITLE
Add automatic installation and instructions

### PR DESCRIPTION
**Pull request by tiangolo**
I added a simple `setup.sh` file that would do all the installation automatically.

And updated the README with the instructions to install and setup everything with just one line to copy-paste.

### CODE REVIEW

**Comment by tiangolo** [comment]
Please let me know if you are OK with this and I'll update the URL to point to your repo instead of mine.

Also, have you considered adding this as a plugin in https://github.com/robbyrussell/oh-my-zsh ?

**Comment by esc** [comment]
@tiangolo I just read the script you provides and there are a few things that I don't agree with entirely and I will leave some comments shortly. As for oh-my-zsh - I don't use this myself, but the code here is WTFPL, so feel free to do as you like. One thing to note when copying code across the internet, it would be nice if this repository stayed the authoritative source for the completion. I am a bit worried the oh-my-zsh version and this one might diverge in terms of available features and completeness.

Also, I did just look for oh-my-zsh and there are several PRs open regarding conda:

https://github.com/robbyrussell/oh-my-zsh/pulls?utf8=%E2%9C%93&q=conda

So I am not too sure that project is open to accepting conda related pull-requests. Anyway, I think the best thing to do would be to improve the documentation here and allow for easier and simpler installation of the plugin under regular zsh and with oh-my-zsh plugins.

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#! /usr/bin/env bash
+
+OLD_PWD=$(pwd)
+
+# Additional ZShell configs
```
Comment (esc) [inline setup.sh]: Not sure what this does, doesn't seem to be related to installation of the completion?

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#! /usr/bin/env bash
```
Comment (esc) [inline setup.sh]: Should this not be `#!/bin/zsh`?

File: setup.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+OLD_PWD=$(pwd)
```
Comment (esc) [inline setup.sh]: This isn't needed, see below.

File: setup.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ADDITIONAL_CONFIGS='source ~/.config/zshrc/*'
+if [[ ! $( grep "$ADDITIONAL_CONFIGS" ~/.zshrc ) ]] ; then
+  echo "Adding custom additional configs from $ADDITIONAL_CONFIGS to ~/.zshrc" ;
+  echo $ADDITIONAL_CONFIGS >> ~/.zshrc ;
+fi ;
+
+# Conda-zsh-completion
+echo 'Installing / updating conda-zsh-completion'
+git clone https://github.com/esc/conda-zsh-completion.git ~/.conda-zsh-completion
+cd ~/.conda-zsh-completion
```
Comment (esc) [inline setup.sh]: There is no need to `cd` and `git pull` during installation, no new changes will arrive.

File: setup.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# Additional ZShell configs
+ADDITIONAL_CONFIGS='source ~/.config/zshrc/*'
+if [[ ! $( grep "$ADDITIONAL_CONFIGS" ~/.zshrc ) ]] ; then
+  echo "Adding custom additional configs from $ADDITIONAL_CONFIGS to ~/.zshrc" ;
+  echo $ADDITIONAL_CONFIGS >> ~/.zshrc ;
+fi ;
+
+# Conda-zsh-completion
+echo 'Installing / updating conda-zsh-completion'
+git clone https://github.com/esc/conda-zsh-completion.git ~/.conda-zsh-completion
```
Comment (esc) [inline setup.sh]: This will clone master, which may not be stable. Probably better to find a way to get the latest release tag and then install from that.

File: setup.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+cd ~/.conda-zsh-completion
+git pull
+echo 'Setting up conda zsh completion configurations'
+echo 'fpath+=~/.conda-zsh-completion' > ~/.config/zshrc/conda-zsh-completion.sh
+echo 'compinit conda' >> ~/.config/zshrc/conda-zsh-completion.sh
+
+# Reload current config
+source ~/.config/zshrc/conda-zsh-completion.sh
+
+cd $OLD_PWD
```
Comment (esc) [inline setup.sh]: This is also not needed.

File: setup.sh [hunk 7]
```diff
@@ -0,0 +1,24 @@
+#! /usr/bin/env bash
+
+OLD_PWD=$(pwd)
+
+# Additional ZShell configs
+ADDITIONAL_CONFIGS='source ~/.config/zshrc/*'
+if [[ ! $( grep "$ADDITIONAL_CONFIGS" ~/.zshrc ) ]] ; then
+  echo "Adding custom additional configs from $ADDITIONAL_CONFIGS to ~/.zshrc" ;
+  echo $ADDITIONAL_CONFIGS >> ~/.zshrc ;
```
Comment (esc) [inline setup.sh]: There are few things worse than scripts that mess with my `.zshrc`. But then again, YMMV.

File: setup.sh [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+echo 'Installing / updating conda-zsh-completion'
+git clone https://github.com/esc/conda-zsh-completion.git ~/.conda-zsh-completion
+cd ~/.conda-zsh-completion
+git pull
+echo 'Setting up conda zsh completion configurations'
+echo 'fpath+=~/.conda-zsh-completion' > ~/.config/zshrc/conda-zsh-completion.sh
+echo 'compinit conda' >> ~/.config/zshrc/conda-zsh-completion.sh
+
+# Reload current config
+source ~/.config/zshrc/conda-zsh-completion.sh
```
Comment (esc) [inline setup.sh]: This will only setup the completion in the current shell. What will happen when the user opens a new terminal?

**Comment by esc** [comment]
Solved by #17

**Comment by tiangolo** [comment]
Thanks for the review. Sadly I didn't have the time to come back to this.

But I'm glad the docs are updated now.

Just for completeness, in case someone else might benefit from it:

As part of my initial Zsh setup I have a script that creates a directory with additonal Zsh sources to load on start:

```bash
# Set additional source files
mkdir ~/.zsh-sources/
echo '# Custom additional source files' >> ~/.zshrc
echo 'for file in ~/.zsh-sources/* ; do source $file; done' >> ~/.zshrc
```

Then to install this completion package I have another script with:

```bash
git clone https://github.com/esc/conda-zsh-completion ~/.zsh/conda-zsh-completion
echo 'fpath=(~/.zsh/conda-zsh-completion $fpath)' > ~/.zsh-sources/conda-zsh-completion.zsh
```
